### ISSUE TITLE
Disabled Google Analytics Setting

### ISSUE THREAD

**Issue by kspearrin** [bitwarden/browser#27]
We use google analytics to to better learn how the extension is being used by users so that improvements can be made. Some users to not want to be tracked in this way. Add an option in settings to allow a user to opt-out of of google analytics.

**Comment by curtiszimmerman** [comment]
So what's the status on this issue? There is still no option to disable analytics. Why was it closed? It seems like you've just deflected the concerns of your users and then quietly closed the issue. Am I wrong?

**Comment by kspearrin** [comment]
This option was added to the extension back in October with the 1.1.1 release. https://github.com/bitwarden/browser/releases/tag/v1.1.1

You can find the setting under Settings > Features

**Comment by hovancik** [comment]
Hi, what exactly is being tracked? 

`We use google analytics to to better learn how the extension is being used by users` is kind of non-telling.

**Comment by kspearrin** [comment]
Use use google analytics in a pretty traditional way: to track page views within the extension and certain events as they happen. For example, we collect how often the autofill feature is used through the current tab method vs the context menu method. This helps us determine which features are being used more often and how they are being used so that we can better make design decisions going forward.

**Comment by fancsali** [comment]
Sorry for being so blunt, but I believe, in order to really make me trust your app, you'd want to remove google analytics completely. At least that's my view, and based on issue #76, I am not alone...

**Comment by Shifterovich** [comment]
@kspearrin Just letting you know, we removed BitWarden from privacytools.io (https://github.com/privacytoolsIO/privacytools.io/issues/719). Opt-in telemetry is okay, but privacy-related applications shouldn't have something like Google Analytics enabled by default.

Please reconsider this.

**Comment by kspearrin** [comment]
@Shifterovich Thanks for the heads up.

It turns out that the analytics weren't as insightful as I had hoped and didn't really end up getting used much. I've just removed them from all Bitwarden client applications for their next releases.

Ref:
- https://github.com/bitwarden/jslib/commit/f4c4f2802637f775c3bfba2ef4109350bc842b8f
- https://github.com/bitwarden/desktop/commit/071074632acb6ccdbf6b51f260b314e443a3b191
- https://github.com/bitwarden/browser/commit/b6705bc338bb0ba0aa7421bcc3f467c43b25f874
- https://github.com/bitwarden/mobile/commit/5b6becc63f8c74fdd7f39ce7eb0bed7d5aaeaba9

**Comment by ibihim** [comment]
Sorry, but the tendency is clear. If the users are not watchful... just the imagination someone thought this is a good idea... cancelling subscription. But you know, right. Google analytics.

**Comment by jeythekey** [comment]
Hi @kspearrin with ping to @Shifterovich,

being concerned about software tracking its users I was feeling really happy and kind of relieved reading your last post, because asides from the user tracking stuff I really love bitwarden.  
I have, however, still some concerns about the use of [Angulartics2/Angulartics2GoogleAnalytics](https://medium.com/altar-io/powerful-analytics-tool-for-angular-applications-angulartics2-c509e17db2a3) code being injected all over the apps. Could you please elaborate shortly in which respect the angulartics approach (which seems to have been there from the first commit on) differs from using ga directly (which you removed)? Did the referenced commits, which apparently "shut up" the core ga-code also make all of the angulartics2-calls dysfunctional? Or what else justifies the statement, that google analytics was removed? 

Simply put: Is there still any user data automatically being sent to google by the bitwarden apps?

That would be really appreciated.

Thanks for your understanding and the fantastic work!

**Comment by kspearrin** [comment]
@jeythekey

> Did the referenced commits, which apparently "shut up" the core ga-code also make all of the angulartics2-calls dysfunctional?

Yes.

All of the Angularlytics stuff are just helpers that end up calling into the `ga()` function that you see was disabled here: https://github.com/bitwarden/jslib/commit/f4c4f2802637f775c3bfba2ef4109350bc842b8f

Essentially Angularlytics has now been "black holed". There is nothing in the apps that is connected to or makes calls to Google Analytics. You could verify this by monitoring your network tab in the browser's dev tools.

**Comment by jeythekey** [comment]
Waoh, thank you so much for clarifying this so quickly!! I suspected that but was not sure, if it might work somehow in parallel. Really happy now :)

**Comment by fancsali** [comment]
This is great news, however https://github.com/bitwarden/jslib/commit/f4c4f2802637f775c3bfba2ef4109350bc842b8f seems to only add a return statement.

I was just wondering, whether the code could be completely removed...
